apis/meta/v1/conditions.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+const (
+	// ReadyCondition is the name of the Ready condition
+	// implemented by all toolkit resources.
+	ReadyCondition string = "Ready"
```

Comment:
Let’s move the Condition type here as well

Reply:
There is a side-effect to this: it will no longer show up on the auto generated API documentation (e.g. https://toolkit.fluxcd.io/components/source/api/#source.toolkit.fluxcd.io/v1alpha1.SourceCondition) but becomes a link to the godoc.org page for the struct.

Reply:
When we will move to the Condition type from 1.19 will it show up in API docs?

Reply:
No, same story.

Reply:
Ok then I don't see why we would keep that type duplicated in each controller, as we will switch to 1.19 eventually.